core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2719,6 +2732,111 @@ object ZIO extends ZIOCompanionPlatformSpecific {
   ): ZIO[R, E, A] =
     new ZIO.EffectAsync(register, blockingOn)
 
+  /**
+   * Imports a synchronous effect that does blocking IO into a pure value.
+   */
+  def effectBlocking[A](effect: => A): Task[A] =
+    blocking(ZIO.effect(effect))
+
+  /**
+   * Imports a synchronous effect that does blocking IO into a pure value,
+   * with a custom cancel effect.
+   *
+   * If the returned `ZIO` is interrupted, the blocked thread running the
+   * synchronous effect will be interrupted via the cancel effect.
+   */
+  def effectBlockingCancelable[A](effect: => A)(cancel: UIO[Unit]): Task[A] =
+    blocking(ZIO.effect(effect)).fork.flatMap(_.join).onInterrupt(cancel)
+
+  /**
+   * Imports a synchronous effect that does blocking IO into a pure value,
+   * refining the error type to `[[java.io.IOException]]`.
+   */
+  def effectBlockingIO[A](effect: => A): IO[IOException, A] =
+    effectBlocking(effect).refineToOrDie[IOException]
+
+  /**
+   * Imports a synchronous effect that does blocking IO into a pure value.
+   *
+   * If the returned `ZIO` is interrupted, the blocked thread running the
+   * synchronous effect will be interrupted via `Thread.interrupt`.
+   *
+   * Note that this adds significant overhead. For performance sensitive
+   * applications consider using `effectBlocking` or
+   * `effectBlockingCancel`.
+   */
+  def effectBlockingInterrupt[A](effect: => A): Task[A] =
+    // Reference user's lambda for the tracer
+    ZIOFn.recordTrace(() => effect) {
+      ZIO.effectSuspendTotal {
+        import java.util.concurrent.atomic.AtomicReference
```

Comment:
Is this fine in `shared`?

Reply:
ah yeah looks like it https://github.com/scala-js/scala-js/blob/8a0655b2ddf3047c13d49814c4debb7f10b88931/javalanglib/src/main/scala/java/lang/Thread.scala